lib/newman_conway.rb

```diff
@@ -1,7 +1,21 @@
 
 
-# Time complexity: ?
-# Space Complexity: ?
+# Time complexity: O(n)
+# Space Complexity: O(n)
 def newman_conway(num)
-  raise NotImplementedError, "newman_conway isn't implemented"
-end
\ No newline at end of file
+  raise ArgumentError if num == 0
+  return "1" if num == 1
+  result = "1 1"
+  n = 3
+  while n <= num
+    conway_num = newman_recursive(n)
+    result += " " + conway_num.to_s
+    n += 1
+  end
+  return result
+end
+
+def newman_recursive(n)
+  return 1 if n <= 2
+  return newman_recursive(newman_recursive(n - 1)) + newman_recursive(n - newman_recursive(n - 1))
```

Comment:
Take a look at the fibonacci notes in the textbook curriculum on algorithms.  This is the same approach as the default recursive solution.  Because you're **not** doing dynamic programming here (solving and saving subproblems for use later), you end up solving the same problems over and over again.  This is actually O(3<sup>3</sup>)